### PR TITLE
Add more information to documentation about how to package extensions with Conda

### PR DESCRIPTION
**Pull request by krinsman**
Because JupyterLab extensions can consist of both Python and TypeScript/Javascript code, and may conceivably begin to frequently use other JupyterLab extensions as dependencies in the future, it seems important that tools be made available to use a language-agnostic package manager to distribute JupyterLab extensions.

The only such package manager I know of is conda, so I have attempted to add information to the developer documentation about how one might package their extensions with conda. I also [added a repository to GitHub containing examples](https://github.com/krinsman/conda_recipes/tree/master/Jupyter/labextensions) but did not want to reference that repository within the documentation itself, because it is not an official part of the Jupyter project, and I wanted to avoid the appearance of self-promotion.

My primary concern is that attempting to distribute JupyterLab extensions primarily via NPM and pip separately will lead to a dependency nightmare which creates among other things (1) a lack of reproducibility when using JupyterLab extensions (2) unnecessary difficulty for end users when installing JupyterLab extensions. Since the extensibility of JupyterLab is one of its most important features, it seems very important to me that JupyterLab extensions be as easy to distribute and install as possible, and this seems like a step in the right direction for that.

Note that the templates I added are _not_ fool proof. In particular, there are several examples of where I was unable to get them to work, although these may require only small changes to fix. (Or maybe they require major changes? I don't know either way.) See [this conda build issue](https://github.com/conda/conda-build/issues/3064) and [this other conda build issue](https://github.com/conda/conda-build/issues/3065).

Any feedback or suggestions you may have would be greatly appreciated. I think adding more guidelines about how to distribute JupyterLab extensions as packages to the documentation could end up being greatly beneficial to the JupyterLab ecosystem as a whole, in ways which are impossible to foresee entirely in advance. My goal is to see this project succeed and to encounter as few barriers and obstacles to widespread adoption as possible.

See also: https://github.com/jupyterlab/jupyterlab/issues/5074

### CODE REVIEW

File: packages/theme-light-extension/style/variables.css [hunk 1]
```diff
@@ -104,10 +104,17 @@ all of MD as it is not optimized for dense, information rich UIs.
    */
 
   --jp-ui-font-scale-factor: 1.2;
-  --jp-ui-font-size0: 0.83333em;
+  --jp-ui-font-size0: 1.2em;
```
Comment (blink1073) [inline packages/theme-light-extension/style/variables.css]: Leftover from another branch :wink:

**Comment by blink1073** [comment]
Wow, thanks for taking this on!  I would love for us to have a cookiecutter repo similar to https://github.com/jupyterlab/extension-cookiecutter-ts that includes a python package and ships a conda package.  We could point to that in the docs rather than having the inline examples.

**Comment by bollwyvl** [comment]
Right, pip or conda can already _distribute_ the tarballs. Don't try to
make them _configure and build_ a lab out of them, as no sane package
manager wants to have minutes of _another_ package manager doing stuff
inside their install/uninstall transaction. Conda's only real advantage in
this case is being able to capture the node dependency... The various
lifecycle scripts are toxic, and are there for solving decades-old sins
(`CLASSPATH`), not enabling new ones.

I think the jupyterlab_delux approach mostly gets it right, as it _doesn't_
introduce the node dependency... But the ux around registering a new
command seems like something we should perhaps tackle in
jupyterlab_launcher, with either an entry_point or a config...

I'll do a more thorough review when I get a chance,

**Comment by krinsman** [comment]
@blink1073 You are right about not having inline examples, I think. They take up a lot of space and somewhat distract from the main point/overall picture.

Right now I am working on creating a conda package for a JupyterLab extension which also comes with a notebook server extension, so that seems feasible for me to do in the near future.

Admittedly I am not really familiar with cookiecutter; I did the XKCD JupyterLab tutorial, but I think I basically skipped the part using cookiecutter. This is something I could work on.

**Comment by blink1073** [comment]
Thanks for getting this started, @krinsman!  We're cleaning up PRs at an in-person meeting.  Closing this for now, but this is a great seed for a cookiecutter that would be very useful for extension authors.

**Comment by jasongrout** [comment]
Thanks very much for the care and explanation you put into these instructions!

**Pull request closed without merging**